napari/_qt/qt_main_window.py

```diff
@@ -1594,14 +1634,73 @@ def _screenshot(
                     add_flash_animation(self._qt_viewer._welcome_widget)
             finally:
                 # make sure we always go back to the right canvas size
-                if size is not None or scale is not None:
+                if size is not None or scale is not None or fit_to_data_extent:
                     canvas.size = prev_size
+                if fit_to_data_extent:
+                    camera.center = old_center
+                    camera.zoom = old_zoom
         else:
             img = self._qt_window.grab().toImage()
             if flash:
                 add_flash_animation(self._qt_window)
         return img
 
+    def export_figure(
+        self,
+        path: Optional[str] = None,
+        scale: float = 1,
+        flash=True,
+    ):
+        """Export an image of the full extent of the displayed layer data.
+
+        This function finds a tight boundary around the data, resets the view
+        around that boundary (and, when scale=1, such that 1 captured pixel is
+        equivalent to one data pixel), takes a screenshot, then restores the
+        previous zoom and canvas sizes. For example with one image layer with
+        an image of 256 x 256 pixels and layer.scale=[1,1], with scale=1 the
+        size of the resulting figure will be 256 x 256 pixels. If layer.scale
+        for both dimensions is set to 0.5, it will be 128 x 128 pixels. Currently,
+        only works when 2 dimensions are displayed.
+
+        Parameters
+        ----------
+        path : Optional[str]
```

Comment:
```suggestion
        path : str, optional
```
[numpydoc.readthedocs.io/en/latest/format.html#parameters](https://numpydoc.readthedocs.io/en/latest/format.html#parameters)